### PR TITLE
Updated cycle counts in the pipeline chapter.

### PR DESCRIPTION
**Pull request by silabs-oysteink**

### CODE REVIEW

File: docs/user_manual/source/pipeline.rst [hunk 1]
```diff
@@ -87,11 +83,25 @@ and zero stall on the data-side memory interface.
   |                       | 4 (target is a non-word-aligned      | EX stage and will cause a flush of the IF stage (including  |
   |                       | non-RVC instruction)                 | prefetch buffer) and ID stage.                              |
   +-----------------------+--------------------------------------+-------------------------------------------------------------+
-  | Instruction Fence     | 5                                    | The FENCE.I instruction as defined in 'Zifencei' of the     |
+  | Instruction Fence     | 5                                    | The FENCE[.I] instructions are defined in 'Zifencei' of the |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/pipeline.rst:86]: I assume that FENCE[.I] in the right most column is supposed to cover both fence and fence.i. If so, just add a separate row for the fence instruction as it is not part of Zifencei as implied now.

File: docs/user_manual/source/pipeline.rst [hunk 2]
```diff
@@ -87,11 +83,25 @@ and zero stall on the data-side memory interface.
...
   |                       |                                      | RISC-V specification. Internally it is implemented as a     |
   |                       | 6 (target is a non-word-aligned      | jump to the instruction following the fence. The jump       |
   |                       | non-RVC instruction)                 | performs the required flushing as described above.          |
+  |                       |                                      | A FENCE.I instruction will not complete until the external  |
+  |                       |                                      | handshake has been completed.                               |
   +-----------------------+--------------------------------------+-------------------------------------------------------------+
+  | Zba, Zbb, Zbc, Zbs    | 1                                    | All instructions from Zba, Zbb, Zbc, Zbs take 1 cycle.      |
+  +-----------------------+--------------------------------------+-------------------------------------------------------------+
+  | Zcmt                  | 2                                    | Tablejumps take 2 cycles.                                   |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/pipeline.rst:95]: Tablejumps -> Table jumps

File: docs/user_manual/source/pipeline.rst [hunk 3]
```diff
@@ -87,11 +83,25 @@ and zero stall on the data-side memory interface.
...
+  +-----------------------+--------------------------------------+-------------------------------------------------------------+
+  | Zcmp                  | 2 - 18                               | The number of cycles depend on the number of registers      |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/pipeline.rst:97]: depend -> depends

File: docs/user_manual/source/pipeline.rst [hunk 4]
```diff
@@ -87,11 +83,25 @@ and zero stall on the data-side memory interface.
...
+  |                       |                                      | saved or restored by the instructions.                      |
+  +-----------------------+--------------------------------------+-------------------------------------------------------------+
+  | Zca, Zcb              | 1                                    | Instructions from Zca and Zcv take 1 cycle.                 |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/pipeline.rst:100]: Zcv -> Zcb

File: docs/user_manual/source/pipeline.rst [hunk 5]
```diff
@@ -87,11 +83,25 @@ and zero stall on the data-side memory interface.
...
+  +-----------------------+--------------------------------------+-------------------------------------------------------------+
+  | WFI, WFE              | 1 -                                  | Instructions causing sleep will not retire until wakeup.    |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/pipeline.rst:102]: Is it really 1 - or is it for example 2 - ? Under what scenario can '1' happen?

File: docs/user_manual/source/pipeline.rst [hunk 6]
```diff
@@ -100,7 +110,16 @@ The |corev| experiences a 1 cycle penalty on the following hazards.
 
  * Load data hazard (in case the instruction immediately following a load uses the result of that load)
  * Jump register (jalr) data hazard (in case that a jalr depends on the result of an immediately preceding non-load instruction)
+ * Any implicit CSR read in ID (mret or table jump) while any implicit or explicit CSR access is in the WB stage
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/pipeline.rst:113]: Any implicit CSR read -> An instruction causing an implicit CSR read
any implicit or explicit CSR access -> a CSR access instruction or an instruction causing an implicit CSR access

Same type of conversion for following lines

Comment (Silabs-ArjanB) [inline docs/user_manual/source/pipeline.rst:113]: I know we don't do it consistently, but try to highlight instruction like jalr by quoting them as ``jalr`` now that you are editing this file anyway

File: docs/user_manual/source/pipeline.rst [hunk 7]
```diff
@@ -100,7 +110,16 @@ The |corev| experiences a 1 cycle penalty on the following hazards.
...
+ * Explicit CSR read in EX while any implicit CSR write is in WB
+ * Explicit CSR read in EX while there is a RAW hazard with an explicit CSR write in WB.
 
 The |corev| experiences a 2 cycle penalty on the following hazards.
 
  * Jump register (jalr) data hazard (in case that a jalr depends on the result of an immediately preceding load instruction)
+ * Any implicit CSR read in ID (mret or table jump) while any implicit or explicit CSR access is in the EX stage
+
+.. note::
+  Implicit CSR reads are reads performed by non-CSR instructions or CSR instructions reading multiple CSR values.
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/pipeline.rst:124]: multiple CSR values -> reading CSR values from another CSR

**Review by Silabs-ArjanB — CHANGES_REQUESTED** [review]

File: docs/user_manual/source/pipeline.rst (hunk 1 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/pipeline.rst]: Fixed

File: docs/user_manual/source/pipeline.rst (hunk 2 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/pipeline.rst]: Fixed

File: docs/user_manual/source/pipeline.rst (hunk 3 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/pipeline.rst]: Fixed

File: docs/user_manual/source/pipeline.rst (hunk 4 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/pipeline.rst]: Fixed

File: docs/user_manual/source/pipeline.rst (hunk 5 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/pipeline.rst]: It should indeed be 2 -, unless we also specify behavior under debug mode which would simply be '1'. Not including debug mode for now.

File: docs/user_manual/source/pipeline.rst (hunk 6 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/pipeline.rst]: Fixed

File: docs/user_manual/source/pipeline.rst (hunk 7 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/pipeline.rst]: Fixed

**Review by Silabs-ArjanB — APPROVED** [review]

**Pull request merged**